Yandex.Metrica destination documentation (#957)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/yandex-metrica.mdx

| RudderStack property | Yandex.Metrica property |
| :--------------| :---------|
| `userId` <br/> `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `UserID` |

## Track

You can use the <Link to="/event-spec/standard-events/track/">`track`</Link> call to send your <Link to="/event-spec/ecommerce-events-spec/">e-commerce events</Link> to Yandex.Metrica.

A sample `track` call is shown below:

```javascript
rudderanalytics.track("Order Completed", {
  order_id: "34897497",
  coupon: "BB20",
  currency: "INR",
  products: [{
      product_id: "43521",
      currency: "INR",
      name: "Rudder Bag",
      price: 3700.99,
      brand: "Nice",
      category: "Bags",
      quantity: 3,
    },
    {
      product_id: "37333",
      currency: "INR",
      name: "RudderStack T-shirt",
      price: 1651.55,
      brand: "Adidas",
      category: "Bags",
      quantity: 3,
    },
  ],
});
```

### Event mapping

The following table lists the default mappings between the RudderStack events and the Yandex.Metrica events:

| RudderStack property | Yandex.Metrica property | Description |
| :--------------| :---------| :-----|

> | `Product Viewed` | `detail` | View the full description of the product. |

Reviewer: ```suggestion
| `Product Viewed` | `detail` | View the full description of the product. |
| `Product List Viewed` | `detail` | View the full description of the product. |
```

### Supported property mappings

The following table lists the properties mapping for Yandex.Metrica's `detail`, `add`, and `remove` events:

| RudderStack property | Yandex.Metrica property |
| :--------------| :---------|
| `properties.product_id` <br/> `properties.sku` <br/> `properties.products[index].product_id` <br/> `properties.products[index].sku` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if name is not present.</span> | `id` |
| `properties.name` <br /> `properties.products[index].name` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if product_id/sku is not present.</span> | `name` |
| `properties.brand` <br /> `properties.products[index].brand` | `brand` |
| `properties.category` <br /> `properties.products[index].category` | `category` |
| `properties.coupon` <br /> `properties.products[index].coupon` | `coupon` |
| `properties.position` <br /> `properties.products[index].position` | `position` |
| `properties.price` <br /> `properties.products[index].price` | `price` |
| `properties.quantity` <br /> `properties.products[index].quantity` | `quantity` |
| `properties.variant` <br /> `properties.products[index].variant` | `variant` |

> The following mappings are applicable only for the `Order Completed` event:

Reviewer: ```suggestion
The following mappings are applicable only for the `purchase` event:
```

  });
```

RudderStack uses Yandex.Metrica's [`hit`](https://yandex.com/support/metrica/objects/hit.html) method to send all the page view data.

### Supported mappings

The following table lists the mappings between the RudderStack attributes and the Yandex.Metrica properties:

| RudderStack property | Yandex.Metrica property |
| :--------------| :---------|
| `context.page.url` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `url` |
| `context.page.title` | `title` |
| `context.page.referrer` | `referer` |

## FAQ

### What is the Yandex.Metrica container name? Where can I find it?

You can set your Yandex.Metrica data container name while setting up your counter for your website, as shown:

<img src="../../assets/event-stream-destinations/yandex-metrica-container-name.png" alt="Yandex.Metrica container name" />

### Where can I find my Yandex.Metrica tag ID?

To obtain your Yandex.Metrica tag ID, follow these steps:

1. Log into your [Yandex.Metrica](https://metrica.yandex.com/) dashboard. 
2. From the left sidebar, click **Visitors**. You can see the tag ID listed in the dashboard as well as in the URL, as shown:

> <img src="../../assets/event-stream-destinations/yandex-metrica-tag-id.png" alt="Yandex.Metrica tag ID" />

Reviewer 2: Should we blur the ID's in the screenshots?

Author: i think that's fine, it's not sensitive info :)

---
title: "Yandex.Metrica"
description: >-
  Step-by-step guide on sending your event data from RudderStack to Yandex.Metrica.
---

[Yandex.Metrica](https://metrica.yandex.com/) is a web analytics platform that lets you get an in-depth understanding of your audience's behavior and use the insights to drive business growth.

<div class="infoBlock">

> Find the open source code for this destination in the <a href="">GitHub repository</a>.

Author: @ujjwal-ab can you share the relevant link for this?

> Find the open source code for this destination in the <a href="">GitHub repository</a>.

Author: ```suggestion
Find the open source code for this destination in the <a href="https://github.com/rudderlabs/rudder-sdk-js/tree/production/integrations/YandexMetrica">GitHub repository</a>.
```

Reviewer: Yeah, it's correct.
